### PR TITLE
doc: add hostbridge emulation doc

### PR DESCRIPTION
**Pull request by gaojunhao0504**
add hostbridge emulation doc and entry

Signed-off-by: Gao Junhao <junhao.gao@intel.com>

### CODE REVIEW

File: doc/developer-guides/hld/hostbridge-virt-hld.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.. _hostbridge_virt_hld:
+
+Hostbridge emulation
+####################
+
+Overview
+********
+
+Hostbridge emulation is based on PCI emulation. However hostbridge emulation only sets PCI configuration space. Device model set the PCI configuration space for hostbridge in SOS, then expose to UOS to detect the PCI hostbridge.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hostbridge-virt-hld.rst]: @NanlinXie should we start to use the terms `User VM` and `Service VM` now that we're going through these documents?

File: doc/developer-guides/hld/hostbridge-virt-hld.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+*****************************
+
+There is PCI host bridge emulation in DM. The bus hierarchy is determined by acrn-dm command line input. Using this command line, as an example::
+
+        acrn-dm -A -m 2048M -c 3 \
+        -s 0:0,hostbridge \
+        -s 1:0,lpc -l com1,stdio \
+        -s 5,virtio-console,@pty:pty_port \
+        -s 3,virtio-blk,b,/data/clearlinux/clearlinux.img \
+        -s 4,virtio-net,tap_LaaG --vsbl /usr/share/acrn/bios/VSBL.bin \
```
Comment (gvancuts) [inline doc/developer-guides/hld/hostbridge-virt-hld.rst]: I would prefer if we could use a script used for the NUCs as an example (so using OVMF.fd, etc.)

File: doc/developer-guides/hld/hostbridge-virt-hld.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+   00:00.0 Host bridge: Network Appliance Corporation Device 1275
+   00:01.0 ISA bridge: Intel Corporation 82371SB PIIX3 ISA [Natoma/Triton II]
+   00:03.0 SCSI storage controller: Red Hat, Inc. Virtio block device
+   00:04.0 Ethernet controller: Red Hat, Inc. Virtio network device
+   00:05.0 Serial controller: Red Hat, Inc. Virtio console
+   i915.enable_guc_loading=0 \
+   i915.enable_hangcheck=0 i915.nuclear_pageflip=1 \
+   i915.enable_guc_submission=0 i915.enable_guc=0" vm1
+
+.. note:: For Clear Linux OS, the lspci command can be installed from the “sysadmin-basic” bundle.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hostbridge-virt-hld.rst]:
```suggestion
.. note:: For Clear Linux OS, the ``lspci`` command can be installed from the ``sysadmin-basic`` bundle.
```

File: doc/developer-guides/hld/hostbridge-virt-hld.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+Overview
+********
+
+Hostbridge emulation is based on PCI emulation. However hostbridge emulation only sets PCI configuration space. Device model set the PCI configuration space for hostbridge in SOS, then expose to UOS to detect the PCI hostbridge.
+
+PCI Host Bridge and hierarchy
+*****************************
+
+There is PCI host bridge emulation in DM. The bus hierarchy is determined by acrn-dm command line input. Using this command line, as an example::
```
Comment (gvancuts) [inline doc/developer-guides/hld/hostbridge-virt-hld.rst]:
```suggestion
There is PCI host bridge emulation in DM. The bus hierarchy is determined by ``acrn-dm`` command line input. Using this command line, as an example::
```

File: doc/developer-guides/hld/hostbridge-virt-hld.rst (hunk 1 above)
Comment (NanlinXie) [reply doc/developer-guides/hld/hostbridge-virt-hld.rst]: @gvancuts , yes, we need use the terms User VM and Service VM now.

Comment (gaojunhao0504) [reply doc/developer-guides/hld/hostbridge-virt-hld.rst]: Thanks, I have updated it.

**Comment by deb-intel** [comment]
@gaojunhao0504 Please resolve the file in conflict.  List "UART emulation in hypervisor" before "Hostbridge emulation"

**Pull request merged**